# ANÁLISIS DEL PROGRESO MUNDIAL DE VACUNACIÓN

##INTRODUCCIÓN
Mediante este proyecto se pretende trabajar con los datos de vacunación por pais obtenidos de la plataforma Kaggle. Se procesaran para poder determinar la cantidad de vacunas aplicadas de cada compañía, obtener la cantidad de vacunas aplicasdas en todo el mundo y calcular el promedio de vacunas aplicadas por país.


##PROCEDIMIENTO

comenzaremos por abrir el archivo de tipo .csv llamado "country_vaccinations"

In [1]:
from google.colab import files
uploaded = files.upload()

Saving country_vaccinations.csv to country_vaccinations.csv


In [2]:
import pandas as pd
vacunas= pd.read_csv("/content/country_vaccinations.csv")
vacunas.info()


<class 'pandas.core.frame.DataFrame'>
RangeIndex: 86512 entries, 0 to 86511
Data columns (total 15 columns):
 #   Column                               Non-Null Count  Dtype  
---  ------                               --------------  -----  
 0   country                              86512 non-null  object 
 1   iso_code                             86512 non-null  object 
 2   date                                 86512 non-null  object 
 3   total_vaccinations                   43607 non-null  float64
 4   people_vaccinated                    41294 non-null  float64
 5   people_fully_vaccinated              38802 non-null  float64
 6   daily_vaccinations_raw               35362 non-null  float64
 7   daily_vaccinations                   86213 non-null  float64
 8   total_vaccinations_per_hundred       43607 non-null  float64
 9   people_vaccinated_per_hundred        41294 non-null  float64
 10  people_fully_vaccinated_per_hundred  38802 non-null  float64
 11  daily_vaccinations_per_milli

Al ver la estructura y tipo de datos de cada columna se puede identificar que la columna date (fecha) se encuentra del tipo object, por lo que será necesario cambiarla a datetime64

In [3]:
vacunas["date"]=pd.to_datetime(vacunas["date"])
vacunas.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 86512 entries, 0 to 86511
Data columns (total 15 columns):
 #   Column                               Non-Null Count  Dtype         
---  ------                               --------------  -----         
 0   country                              86512 non-null  object        
 1   iso_code                             86512 non-null  object        
 2   date                                 86512 non-null  datetime64[ns]
 3   total_vaccinations                   43607 non-null  float64       
 4   people_vaccinated                    41294 non-null  float64       
 5   people_fully_vaccinated              38802 non-null  float64       
 6   daily_vaccinations_raw               35362 non-null  float64       
 7   daily_vaccinations                   86213 non-null  float64       
 8   total_vaccinations_per_hundred       43607 non-null  float64       
 9   people_vaccinated_per_hundred        41294 non-null  float64       
 10  people_ful

###Cantidad de Vacunas aplicadas de cada compañia
Tomamos la columna "total_vaccinations" la cual contiene el numero total de vacunas aplicadas por dia y la columna "vaccines" que especifica el tipo de vacuna que se aplicó.

Para poder determinar la cantidad de vacunas aplicadas de cada compañia agrupamos por "vaccines", lo cual nos arroja el siguiente resultado

In [4]:
vacunas[["total_vaccinations", "vaccines"]].groupby("vaccines").sum()

total_vaccinations
vaccines                                                              
Abdala, Johnson&Johnson, Oxford/AstraZeneca, Pf...        1.784348e+08
Abdala, Moderna, Oxford/AstraZeneca, Pfizer/Bio...        1.791450e+10
Abdala, Sinopharm/Beijing, Sinovac, Soberana02,...        4.228568e+08
Abdala, Soberana Plus, Soberana02                         6.405166e+09
COVIran Barekat, Covaxin, FAKHRAVAC, Oxford/Ast...        1.069480e+10
...                                                                ...
Pfizer/BioNTech, Sinovac, Turkovac                        3.392310e+10
Pfizer/BioNTech, Sputnik V                                3.863380e+06
QazVac, Sinopharm/Beijing, Sputnik V                      3.585429e+09
Sinopharm/Beijing                                         3.080671e+07
Sinopharm/Beijing, Sputnik V                              6.451243e+08

[84 rows x 1 columns]

###Cantidad de vacunas aplicadas en todo el mundo
Para poder obtener esta información, se manda a llamar a la columna "total_vaccinations" que es la que contiene el número total de vacunas aplicadas al día, y le aplicamos la funcion sum().

In [5]:
vacunas["total_vaccinations"].sum()


2002854013761.0

###Promedio de vacunas aplicadas por país

Se seleccionan las columnas "country" y "total vaccinations" y agrupamos por pais para poder obtener el promedio de vacunas que se aplicaron por pais.

In [6]:
vacunas[["country", "total_vaccinations"]].groupby("country").mean()

total_vaccinations
country                              
Afghanistan              2.566118e+06
Albania                  1.191695e+06
Algeria                  9.956086e+06
Andorra                  5.827721e+04
Angola                   6.074066e+06
...                               ...
Wales                    4.132861e+06
Wallis and Futuna        9.142925e+03
Yemen                    4.794049e+05
Zambia                   1.048484e+06
Zimbabwe                 4.181618e+06

[223 rows x 1 columns]

###Cantidad de vacunas aplicadas el dia 29/01/2021 en todo el mundo

En este caso primero se mandó a llamar la columna "date" especificando que queriamos los valores que coincidieran con la fecha "2021-01-29" y lo guardamos en una variable llamada "vacunas_29_01_2021".

Posteriormente de la variable que se guardó, mandamos a llamar la columna "total_vaccinations" y sumamos los valores. Por lo tanto el resultado del total de vacunas aplicadas en todo el mundo en la fecha señalada fue de 82,952,931 vacunas.

In [7]:
vacunas_29_01_21 = vacunas[vacunas['date'] == '2021-01-29']
total_vacunas_29_01_21 = vacunas_29_01_21['total_vaccinations'].sum()
print(f"El total de vacunas aplicadas el 29/01/21 en todo el mundo es: {total_vacunas_29_01_21}")

El total de vacunas aplicadas el 29/01/21 en todo el mundo es: 82952931.0


###Diferenecias de aplicacion
se crea un dataframe donde tenemos una columna llamada "diferencias" en la cual se guardan la diferencia de las cantidades de aplicacion entre "daily_vaccinations" y "daily_vaccinations_raw"

In [8]:
conDiferencias=vacunas
conDiferencias["diferencias"]=conDiferencias["daily_vaccinations"]- conDiferencias["daily_vaccinations_raw"]
conDiferencias

country iso_code       date  total_vaccinations  people_vaccinated  \
0      Afghanistan      AFG 2021-02-22                 0.0                0.0   
1      Afghanistan      AFG 2021-02-23                 NaN                NaN   
2      Afghanistan      AFG 2021-02-24                 NaN                NaN   
3      Afghanistan      AFG 2021-02-25                 NaN                NaN   
4      Afghanistan      AFG 2021-02-26                 NaN                NaN   
...            ...      ...        ...                 ...                ...   
86507     Zimbabwe      ZWE 2022-03-25           8691642.0          4814582.0   
86508     Zimbabwe      ZWE 2022-03-26           8791728.0          4886242.0   
86509     Zimbabwe      ZWE 2022-03-27           8845039.0          4918147.0   
86510     Zimbabwe      ZWE 2022-03-28           8934360.0          4975433.0   
86511     Zimbabwe      ZWE 2022-03-29           9039729.0          5053114.0   

       people_fully_vaccinated  daily_vaccinations_raw  daily_vaccinations  \
0                          NaN                     NaN                 NaN   
1                          NaN                     NaN              1367.0   
2                          NaN                     NaN              1367.0   
3                          NaN                     NaN              1367.0   
4                          NaN                     NaN              1367.0   
...                        ...                     ...                 ...   
86507                3473523.0                139213.0             69579.0   
86508                3487962.0                100086.0             83429.0   
86509                3493763.0                 53311.0             90629.0   
86510                3501493.0                 89321.0            100614.0   
86511                3510256.0                105369.0            103751.0   

       total_vaccinations_per_hundred  people_vaccinated_per_hundred  \
0                                0.00                           0.00   
1                                 NaN                            NaN   
2                                 NaN                            NaN   
3                                 NaN                            NaN   
4                                 NaN                            NaN   
...                               ...                            ...   
86507                           57.59                          31.90   
86508                           58.25                          32.38   
86509                           58.61                          32.59   
86510                           59.20                          32.97   
86511                           59.90                          33.48   

       people_fully_vaccinated_per_hundred  daily_vaccinations_per_million  \
0                                      NaN                             NaN   
1                                      NaN                            34.0   
2                                      NaN                            34.0   
3                                      NaN                            34.0   
4                                      NaN                            34.0   
...                                    ...                             ...   
86507                                23.02                          4610.0   
86508                                23.11                          5528.0   
86509                                23.15                          6005.0   
86510                                23.20                          6667.0   
86511                                23.26                          6874.0   

                                                vaccines  \
0      Johnson&Johnson, Oxford/AstraZeneca, Pfizer/Bi...   
1      Johnson&Johnson, Oxford/AstraZeneca, Pfizer/Bi...   
2      Johnson&Johnson, Oxford/AstraZeneca, Pfizer/Bi...   
3      Johnson&Johnson, Oxford/AstraZeneca, Pfizer/Bi...   

###Periodo de tiempo de vacunanción
vamos a obtener el periodo de tiempo entre el registro con la fecha mas reciente y el registro con la fecha más antigua, para esto obtendremos el valor minimo y máximo de la columna "date" para posteriormente restarlo.

In [9]:
fecha_minima = vacunas['date'].min()
fecha_maxima = vacunas['date'].max()
periodo_tiempo = fecha_maxima - fecha_minima
print(f"La fecha más antigua en la base de datos es: {fecha_minima}")
print(f"La fecha más reciente en la base de datos es: {fecha_maxima}")
print(f"El período de tiempo de vacunación es: {periodo_tiempo}")

La fecha más antigua en la base de datos es: 2020-12-02 00:00:00
La fecha más reciente en la base de datos es: 2022-03-29 00:00:00
El período de tiempo de vacunación es: 482 days 00:00:00


###Cantidad de Vacunas utilizadas cada día
A continuación se crea una nueva columna donde se toma "vaccines" para poder saber que cantidad de vacunas se utilizó cada día del registro.

In [10]:
conCantidad=vacunas
conCantidad['canVac'] = conCantidad['vaccines'].str.split(',').apply(len)
conCantidad

country iso_code       date  total_vaccinations  people_vaccinated  \
0      Afghanistan      AFG 2021-02-22                 0.0                0.0   
1      Afghanistan      AFG 2021-02-23                 NaN                NaN   
2      Afghanistan      AFG 2021-02-24                 NaN                NaN   
3      Afghanistan      AFG 2021-02-25                 NaN                NaN   
4      Afghanistan      AFG 2021-02-26                 NaN                NaN   
...            ...      ...        ...                 ...                ...   
86507     Zimbabwe      ZWE 2022-03-25           8691642.0          4814582.0   
86508     Zimbabwe      ZWE 2022-03-26           8791728.0          4886242.0   
86509     Zimbabwe      ZWE 2022-03-27           8845039.0          4918147.0   
86510     Zimbabwe      ZWE 2022-03-28           8934360.0          4975433.0   
86511     Zimbabwe      ZWE 2022-03-29           9039729.0          5053114.0   

       people_fully_vaccinated  daily_vaccinations_raw  daily_vaccinations  \
0                          NaN                     NaN                 NaN   
1                          NaN                     NaN              1367.0   
2                          NaN                     NaN              1367.0   
3                          NaN                     NaN              1367.0   
4                          NaN                     NaN              1367.0   
...                        ...                     ...                 ...   
86507                3473523.0                139213.0             69579.0   
86508                3487962.0                100086.0             83429.0   
86509                3493763.0                 53311.0             90629.0   
86510                3501493.0                 89321.0            100614.0   
86511                3510256.0                105369.0            103751.0   

       total_vaccinations_per_hundred  people_vaccinated_per_hundred  \
0                                0.00                           0.00   
1                                 NaN                            NaN   
2                                 NaN                            NaN   
3                                 NaN                            NaN   
4                                 NaN                            NaN   
...                               ...                            ...   
86507                           57.59                          31.90   
86508                           58.25                          32.38   
86509                           58.61                          32.59   
86510                           59.20                          32.97   
86511                           59.90                          33.48   

       people_fully_vaccinated_per_hundred  daily_vaccinations_per_million  \
0                                      NaN                             NaN   
1                                      NaN                            34.0   
2                                      NaN                            34.0   
3                                      NaN                            34.0   
4                                      NaN                            34.0   
...                                    ...                             ...   
86507                                23.02                          4610.0   
86508                                23.11                          5528.0   
86509                                23.15                          6005.0   
86510                                23.20                          6667.0   
86511                                23.26                          6874.0   

                                                vaccines  \
0      Johnson&Johnson, Oxford/AstraZeneca, Pfizer/Bi...   
1      Johnson&Johnson, Oxford/AstraZeneca, Pfizer/Bi...   
2      Johnson&Johnson, Oxford/AstraZeneca, Pfizer/Bi...   
3      Johnson&Johnson, Oxford/AstraZeneca, Pfizer/Bi...   

###Registros antes del 20 de diciembre de 2020
Se selecciona del dataframe solamente los registros de vacunación aplicadas antes del 20 de diciembre de 2020 para darnos una idea de cuantas vacunas se aplicaron recién iniciadas las campañas de vacunación a nivel mundial.

In [11]:
antes20 = vacunas[vacunas['date'] < '2020-12-20']
antes20

country iso_code       date  total_vaccinations  \
13403         Canada      CAN 2020-12-14                 5.0   
13404         Canada      CAN 2020-12-15               727.0   
13405         Canada      CAN 2020-12-16              3025.0   
13406         Canada      CAN 2020-12-17              7279.0   
13407         Canada      CAN 2020-12-18             11296.0   
...              ...      ...        ...                 ...   
82362  United States      USA 2020-12-15             84638.0   
82363  United States      USA 2020-12-16            244549.0   
82364  United States      USA 2020-12-17            517161.0   
82365  United States      USA 2020-12-18            933551.0   
82366  United States      USA 2020-12-19           1115437.0   

       people_vaccinated  people_fully_vaccinated  daily_vaccinations_raw  \
13403                5.0                      NaN                     NaN   
13404              727.0                      NaN                   722.0   
13405             3025.0                      NaN                  2298.0   
13406             7279.0                      NaN                  4254.0   
13407            11296.0                      NaN                  4017.0   
...                  ...                      ...                     ...   
82362            76984.0                   6281.0                 49771.0   
82363           231496.0                   6757.0                159911.0   
82364           496980.0                   7507.0                272612.0   
82365           904656.0                   8718.0                416390.0   
82366          1082608.0                   9778.0                181886.0   

       daily_vaccinations  total_vaccinations_per_hundred  \
13403                 NaN                            0.00   
13404               722.0                            0.00   
13405              1510.0                            0.01   
13406              2425.0                            0.02   
13407              2823.0                            0.03   
...                   ...                             ...   
82362             27175.0                            0.03   
82363             71420.0                            0.07   
82364            121718.0                            0.16   
82365            180653.0                            0.28   
82366            180858.0                            0.34   

       people_vaccinated_per_hundred  people_fully_vaccinated_per_hundred  \
13403                           0.00                                  NaN   
13404                           0.00                                  NaN   
13405                           0.01                                  NaN   
13406                           0.02                                  NaN   
13407                           0.03                                  NaN   
...                              ...                                  ...   
82362                           0.02                                  0.0   
82363                           0.07                                  0.0   
82364                           0.15                                  0.0   
82365                           0.27                                  0.0   
82366                           0.33                                  0.0   

       daily_vaccinations_per_million  \
13403                             NaN   
13404                            19.0   
13405                            40.0   
13406                            64.0   
13407                            74.0   
...                               ...   
82362                            82.0   
82363                           215.0   
82364                           367.0   
82365                           544.0   
82366                           545.0   

                                                vaccines  \
13403  Johnson&Johnson, Moderna, Oxford/AstraZeneca, ...   
13404  Johnson&Johnson, Moderna, Oxford/AstraZen

###Vacunas pfizer
Se crea un nuevo dataframe denominado Pfizer donde se guardan los datos donde la vacuna aplicada fue de la farmaceutica pfizer.

In [12]:
pfizer = vacunas[vacunas['vaccines'].str.contains('Pfizer')]
pfizer

country  iso_code       date  total_vaccinations  \
0      Afghanistan       AFG 2021-02-22                 0.0   
1      Afghanistan       AFG 2021-02-23                 NaN   
2      Afghanistan       AFG 2021-02-24                 NaN   
3      Afghanistan       AFG 2021-02-25                 NaN   
4      Afghanistan       AFG 2021-02-26                 NaN   
...            ...       ...        ...                 ...   
85069        Wales  OWID_WLS 2022-03-24           6921195.0   
85070        Wales  OWID_WLS 2022-03-25           6923298.0   
85071        Wales  OWID_WLS 2022-03-26           6923706.0   
85072        Wales  OWID_WLS 2022-03-27           6925183.0   
85073        Wales  OWID_WLS 2022-03-28           6927437.0   

       people_vaccinated  people_fully_vaccinated  daily_vaccinations_raw  \
0                    0.0                      NaN                     NaN   
1                    NaN                      NaN                     NaN   
2                    NaN                      NaN                     NaN   
3                    NaN                      NaN                     NaN   
4                    NaN                      NaN                     NaN   
...                  ...                      ...                     ...   
85069          2526832.0                2397677.0                  1756.0   
85070          2527100.0                2398097.0                  2103.0   
85071          2527184.0                2398322.0                   408.0   
85072          2528051.0                2398676.0                  1477.0   
85073          2528664.0                2399341.0                  2254.0   

       daily_vaccinations  total_vaccinations_per_hundred  \
0                     NaN                            0.00   
1                  1367.0                             NaN   
2                  1367.0                             NaN   
3                  1367.0                             NaN   
4                  1367.0                             NaN   
...                   ...                             ...   
85069              1705.0                          218.33   
85070              1667.0                          218.40   
85071              1519.0                          218.41   
85072              1505.0                          218.46   
85073              1609.0                          218.53   

       people_vaccinated_per_hundred  people_fully_vaccinated_per_hundred  \
0                               0.00                                  NaN   
1                                NaN                                  NaN   
2                                NaN                                  NaN   
3                                NaN                                  NaN   
4                                NaN                                  NaN   
...                              ...                                  ...   
85069                          79.71                                75.64   
85070                          79.72                                75.65   
85071                          79.72                                75.66   
85072                          79.75                                75.67   
85073                          79.77                                75.69   

       daily_vaccinations_per_million  \
0                                 NaN   
1                                34.0   
2                                34.0   
3                                34.0   
4                                34.0   
...                               ...   
85069                           538.0   
85070                           526.0   
85071                           479.0   
85072                           475.0   
85073                           508.0   

                                                vaccines  \
0      Johnson&Johnson, Oxford/AstraZeneca, Pfizer/Bi...   
1      Johnson&Johnson, Oxford/AstraZeneca, Pfizer/Bi...  

###Almacenamiento de dataframes
los dataframes con los que estuvimos trabajando en este notebook se guardaran en un libro de excel denominado "resultadosReto"

In [15]:
nombre_archivo = 'resultadosReto.xlsx'

# Crear un escritor de Excel
with pd.ExcelWriter(nombre_archivo, engine='openpyxl') as writer:
    # Almacenar cada DataFrame en una hoja separada
    conDiferencias.to_excel(writer, sheet_name='conDiferencias', index=False)
    conCantidad.to_excel(writer, sheet_name='conCantidad', index=False)
    antes20.to_excel(writer, sheet_name='antes20', index=False)
    pfizer.to_excel(writer, sheet_name='pfizer', index=False)

print(f'Los DataFrames se han guardado en el archivo {nombre_archivo}.')

Los DataFrames se han guardado en el archivo resultadosReto.xlsx.


Movemos el archivo para para poderlo visualizar en google drive.

In [16]:
from google.colab import drive
drive.mount('/content/drive')

Mounted at /content/drive


In [17]:
import shutil
shutil.move('resultadosReto.xlsx', '/content/drive/MyDrive/')

'/content/drive/MyDrive/resultadosReto.xlsx'

##CONCLUSIÓN
Esta base de datos obtenida en Kaggle nos hace darnos una idea de que tanto ha avanzado la vacunación a nivel mundial a lo largo de los 482 días que contemplan los datos; de igual forma pudimos ver y analizar la estructura y tipos de datos que se registraron, transformarlos y saber cómo trabajar con cada uno de ellos.

Es importante obtener un resumen de esta información porque así podemos visualizar que país necesita más apoyo en sus campañas de vacunación, que vacunas son las más demandadas y que porcentaje de la población mundial ha recibido al menos una dosis.

